src/BlockTypes/ProductQuery.php

```diff
@@ -117,4 +135,71 @@ private function get_on_sale_products_query( $variation_props ) {
 			),
 		);
 	}
+
+	/**
+	 * Set the query vars that are used by filter blocks.
+	 *
+	 * @param array $qvars Public query vars.
+	 * @return array
+	 */
+	public function set_query_vars( $qvars ) {
+		$filter_query_args = array( PriceFilter::MIN_PRICE_QUERY_VAR, PriceFilter::MAX_PRICE_QUERY_VAR );
+		return array_merge( $qvars, $filter_query_args );
+	}
+
+	/**
+	 * Return queries that are generated by query args
+	 *
+	 * @return array
+	 */
+	private function get_queries_by_applied_filters() {
+		return array( 'price_filter' => $this->get_filter_by_price_query() );
+	}
+
+	/**
+	 * Return queries that are generated by attributes
+	 *
+	 * @param array $variation_props Dedicated attributes for the variation.
+	 * @return array
+	 */
+	private function get_queries_by_attributes( $variation_props ) {
+		return array(
+			'on_sale' => ( ! isset( $variation_props['attributes']['query']['onSale'] ) || true !== $variation_props['attributes']['query']['onSale'] ) ? array() : $this->get_on_sale_products_query(),
+		);
```

Comment:
I know it's not introduced in this PR, but I wonder if we could extract this check into its own variable to make the code easier to read, something like:

```suggestion
		$on_sale = ! isset( $variation_props['attributes']['query']['onSale'] ) || true !== $variation_props['attributes']['query']['onSale'];
		return array(
			'on_sale' => $on_sale ? array() : $this->get_on_sale_products_query(),
		);
```